osgtest/tests/test_155_stashcache.py

```diff
@@ -213,3 +220,24 @@ def test_04_start_stash_cache(self):
 
     def test_05_start_stash_cache_auth(self):
         start_xrootd("stash-cache-auth")
+
+    def test_06_start_namespaces_json_server(self):
+        # Start the namespaces JSON server in the background
+        # Don't wait for it to finish, but keep track of the process by saving
+        # the process object in core.state.
+        setstate("namespaces_json_server_proc", None)
+        setcfg("STASH_NAMESPACE_URL", "")
+        q_python = shlex.quote(sys.executable)
+        logfile = "/tmp/namespaces_json.log.%d" % os.getpid()
+        setcfg("namespaces_json_server_logfile", logfile)
+        proc = Popen(f"{q_python} -m osgtest.library.namespaces_json_server > {logfile} 2>&1", shell=True)
+
+        # Make sure it didn't immediately crash
+        time.sleep(2)
+        ret = proc.poll()
+        if ret is not None:
+            core.system(["/bin/cat", logfile])
+            self.assertEqual(ret, None, f"namespaces JSON server terminated prematurely with code {ret}")
+
+        setstate("namespaces_json_server_proc", proc)
+        setcfg("STASH_NAMESPACE_URL", "http://localhost:1080")
```

Comment:
I can see this test being generally useful in the future. It's a good candidate for breaking out into one of the `osg-test` libs!

Reply:
I'll break it out if I use it again; I probably could have gotten rid of most of this if I had just made a systemd .service file for namespaces_json_server

Reply:
> I'll break it out if I use it again

Yeah, that's what I was thinking

> I probably could have gotten rid of most of this if I had just made a systemd .service file for namespaces_json_server

I did have the same thought here. It's probably less work to just create the systemd .service, isn't it? If you agree, let's just go with this implementation now and add a `# TODO` comment to turn this into a systemd service later

Reply:
OK, TODOs added.